composite/token-allocation/src/adapter.ts

```diff
@@ -35,19 +36,34 @@ export const marketCapTotalValue = (
     .toNumber()
 }
 
-const inputParams = {
-  allocations: true,
-  quote: false,
-}
-
 const toValidAllocations = (allocations: TokenAllocations): TokenAllocations => {
   if (!allocations.every((t) => !!t.symbol))
     throw new AdapterError({ message: `Symbol not available for all tokens.`, statusCode: 400 })
-  return allocations.map((t) => ({
-    symbol: t.symbol.toUpperCase(),
-    decimals: t.decimals || DEFAULT_TOKEN_DECIMALS,
-    balance: t.balance || DEFAULT_TOKEN_BALANCE * 10 ** (t.decimals || DEFAULT_TOKEN_DECIMALS),
-  }))
+
+  const _toValidDecimals = (decimals: number) =>
```

Comment:
This can be undefined:

```suggestion
  const _toValidDecimals = (decimals: number | undefined) =>
```

Reply:
I'm not sure it can, it's a mandatory field on `TokenAllocation`

Reply:
Yes, on `TokenAllocation` but this is before we have `TokenAllocation`. We are mapping and validating an `any[]` input as `TokenAllocation[]`